### PR TITLE
feat: start implementing computeOperand, by computeDst

### PR DESCRIPTION
**Pull request by Eikix**
petite PR pour bien finir le weekend

### CODE REVIEW

File: src/primitives/uint.ts [hunk 1]
```diff
@@ -29,6 +29,8 @@ export const Uint64ConversionError = {
 export class UnsignedInteger {
   static readonly ZERO: Uint32 = 0 as Uint32;
```
Comment (greged93) [inline src/primitives/uint.ts:30]: Maybe we need to specify the type of the ZERO here as well?

File: src/vm/instruction.ts [hunk 2]
```diff
@@ -74,4 +76,17 @@ enum Opcode {
...
 }
 
-export type Instruction = {};
+export type Instruction = {
+  // The offset to add or sub to ap/fp to obtain the Destination Operand
+  offDst: Int16;
+  offOp0: Int16;
+  offOp1: Int16;
+  dstReg: RegisterFlag;
+  op0Reg: RegisterFlag;
```
Comment (greged93) [inline src/vm/instruction.ts:85]: Should this be `DstRegister` and `Op0Register`?

**Comment by Eikix** [comment]
done!~

File: src/primitives/relocatable.ts [hunk 3]
```diff
@@ -110,3 +110,25 @@ export class Relocatable {
     return this.offset;
   }
 }
+
+/**
+ * Subclass of Relocatable, specific to the Allocation Pointer (Ap) and the Frame Pointer (Fp)
+ * These CairoVM registers are considered relocatables must can only have segment index equal to 1
+ * as they always point to the execution segment.
+ */
+export class VmPointer extends Relocatable {
```
Comment (greged93) [inline src/primitives/relocatable.ts:119]: Should we rename this the ExecutionSegment? (Cairo paper page 41)

File: src/primitives/relocatable.ts [hunk 4]
```diff
@@ -110,3 +110,25 @@ export class Relocatable {
     return this.offset;
   }
 }
+
+/**
+ * Subclass of Relocatable, specific to the Allocation Pointer (Ap) and the Frame Pointer (Fp)
+ * These CairoVM registers are considered relocatables must can only have segment index equal to 1
```
Comment (greged93) [inline src/primitives/relocatable.ts:116]:
```suggestion
 * These CairoVM registers are considered relocatables must only have segment index equal to 1
```

File: src/primitives/relocatable.ts [hunk 5]
```diff
@@ -110,3 +110,25 @@ export class Relocatable {
...
+    super(1, offset);
+  }
+}
+
+/**
+ * Subclass of Relocatable, specific to the Program Counter (PC).
+ * PC points to the program segment. Its segment will be 0 until the last instruction.
+ * At the end of a program run, the PC will be set to the end pointer, i.e. the address of the end segment.
+ */
+export class ProgramCounter extends Relocatable {
```
Comment (greged93) [inline src/primitives/relocatable.ts:130]: ProgramSegment? (Cairo paper page 41)

**Review by greged93 — COMMENTED** [review]
typo + 2 comments

File: src/primitives/relocatable.ts (hunk 3 above)
Comment (Eikix) [reply src/primitives/relocatable.ts:119]: but it's a relocatable no? the execution segment is the segment that has index 1?

File: src/primitives/relocatable.ts (hunk 5 above)
Comment (Eikix) [reply src/primitives/relocatable.ts:130]: I feel like the ProgramCounter lives in the ProgramSegment so to speak

File: src/primitives/relocatable.ts (hunk 3 above)
Comment (greged93) [reply src/primitives/relocatable.ts:119]: Mmmmm true, I just find the naming a little strange

Comment (Eikix) [reply src/primitives/relocatable.ts:119]: Hmm true, 
`FpOrAp`?
MemoryPointer?
Registers?

Comment (greged93) [reply src/primitives/relocatable.ts:119]: MemoryPointer or FpOrAp sounds good, preferences for MemoryPointer personally.

Comment (Eikix) [reply src/primitives/relocatable.ts]: ok

File: src/run-context/runContext.ts [hunk 6]
```diff
@@ -8,8 +12,8 @@ export const PCError = {
 };
 
 export class RunContext {
-  private pc: VmPointer;
-  private ap: VmPointer;
+  private pc: MemoryPointer;
+  private ap: MemoryPointer;
   private fp: ProgramCounter;
```
Comment (greged93) [inline src/run-context/runContext.ts:17]: Shouldn't ap and fp be `MemoryPointer`?, while pc `ProgramCounter`

**Review by greged93 — COMMENTED** [review]
small mistake on the run context, looks good otherwise

File: src/run-context/runContext.ts (hunk 6 above)
Comment (Eikix) [reply src/run-context/runContext.ts]: thx nice catch, fixed!

**Review by greged93 — APPROVED** [review]
lgtm

**Pull request merged**